connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -91,13 +94,15 @@
     private static final Logger log = LoggerFactory.getLogger(DistributedHerder.class);
 
     private static final long RECONFIGURE_CONNECTOR_TASKS_BACKOFF_MS = 250;
+    private static final int BULK_THREAD_POOL_SIZE = 8;
 
     private final Time time;
 
     private final int workerSyncTimeoutMs;
     private final int workerUnsyncBackoffMs;
 
     private final ExecutorService forwardRequestExecutor;
+    private final ExecutorService bulkExecutor;
```

Comment:
I've commented on the `Worker` placement above. My issue is mainly w.r.t. the things that need to happen when we start a connector and herder seems to have better information about those (passing an object with an `init` method that the worker should call before starting the connector seems a bit unnatural - happy to reconsider if you insist).

With respect to naming, it makes me sad how clearly separated are the concerns of "start and stop" when it comes to naming. Have been trying for a while with not great success (godExecutor, motherNatureExecutor, springAutumnExecutor ... )